src/ARAnchorManager.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * A class to manage registration and unregistration of anchors
+ * for any THREE.Object3D. Anchors are world space and handle updating
+ * the position of the object to which one is assigned.
+ * AnchorManager emits events pertaining to anchors.
+ */
+export default class ARAnchorManager extends EventDispatcher {
+  constructor(vrDisplay) {
+    super();
+    if (!(vrDisplay instanceof VRDisplay) || !vrDisplay.getAnchors) {
+      throw new Error("invalid vr display parameter");
+    }
+    this.vrDisplay_ = vrDisplay;
+    this.anchorsToObject3Ds_ = new Map();
+    this.object3DsToAnchors_ = new Map();
+    this.vrDisplay_.addEventListener('anchorsupdated', 
+        event => this.onAnchorsUpdated_(event));
+    // This attribute is used to get the scale value from the anchor model
+    // matrix but not use it. It is just to avoid garbage collection.
+    this.scale_  = new Vector3();
+    // This attribute is used to calculate the matrix to be passed to the
+    // anchor creation code only using the position and quaternion from the
+    // object3d.
+    this.matrix_ = new Matrix4();
+  }
+  /**
+   * Assign a new anchor to an THREE.Object3D.
+   * @param object3d {THREE.Object3D} A Three Object3D instance to create and
+   * manage an anchor for. The current transformation (position/orientation) of
+   * the Object3D will be used to create the anchor and this class will keep
+   * it up to date with any tracking changes detected by the system.
+   * IMPORTANT: The passed Object3D instance should be in world space!
+   * @throws Error if:
+   * - The given object3d is not an instance of Object3D.
+   * 
+   */
+  add(object3d) {
+    if (!(object3d instanceof Object3D)) {
+      throw new Error('Invalid Object3D trying to add an anchor');
+    };
+    // Do not allow to create an anchor for an object3d that already has one.
+    if (this.object3DsToAnchors_.has(object3d)) {
+      return this;
+    }
+    // If the object has a parent and its parent has a parent, it is very
+    // likely that it may not be in world space so warn the developer.
+    // NOTE: Commented out for now because if we use anchors for planes in
+    // ARPlanes, they are added to a identity main parent object but does not
+    // mean they are not in world space. Getting this warning at all times is
+    // not optimal.
+//    if (object3d.parent && object3d.parent.parent) {
+//      console.warn("anchors should be added for Object3D-s in world space");
+//    }
+    // Calculate the model matrix of the object3d using the position and
+    // quaternion of the object.
+    this.scale_.set(1, 1, 1);
+    this.matrix_.compose(object3d.position, object3d.quaternion, this.scale_);
+    const anchor = this.vrDisplay_.addAnchor(this.matrix_.elements);
+    // Store the anchor and the object3d in the structure to retrieve them
+    // later on.
+    this.anchorsToObject3Ds_.set(anchor, object3d);
+    this.object3DsToAnchors_.set(object3d, anchor);
+    return this;
```

Comment:
I think it's weird to return this from add and return a boolean from remove. Maybe just don't return here? If you want to return, at least add a @return doc to the docstring for clarity

Reply:
Will make the remove function to return this too.

Reply:
Not totally sure on best practices with regards to returning 'this', but as commented elsewhere, the boolean return on remove provided some feedback as to whether it was a successful remove. This feedback would be helpful if a user thought they registered an object but didn't. If we're returning 'this' consistently, then maybe we should add a throw for a failed remove. The boolean return for a remove method is consistent with Javascript classes like Map() (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/delete)

Reply:
And maybe worth noting @lincolnfrog for a Map, it returns 'this' on a .set(k,v) call. So there is at least some precedent for different return types. No strong opinions on the subject except that I think there should be feedback for a failed remove because it's a user error.

Reply:
ok, I don't feel strongly about this. Personally, I hate the "return this" paradigm entirely, I think its super weird and non-uniform (for reasons like this - you can chain together add calls but not remove calls because it returns a boolean? WEIRD!). 

I am outvoted by the JS community however, so I don't really have a strong opinion on what you do here. Personally, I never ever ever-in-a-million-years return this and never use the chaining pattern because, as I mentioned, I find it counter-intuitive and completely illegible. It strikes me as a very lazy design pattern that is more cute than useful.

Reply:
Chaining is a common pattern in JS like you said -- fine either way, but should be consistent

Reply:
@jsantell thats the problem - it CAN'T be consistent. Some functions need to actually return something meaningful, such as remove, and others have the "luxury" of returning this so they can be chained. Anyway, put peeve of mine and it's fine to do the standard thing. I would say following the pattern established by Map is a good way to go, so +1 to that idea.

Reply:
Ok, I think following the Map pattern is ok. I will change it and re-push.